pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go

```diff
@@ -0,0 +1,172 @@
+		c.Put(ctx, t.Cockroach(), "./cockroach", roachNodes)
+
+		startOpts := option.DefaultStartOpts()
+		settings := install.MakeClusterSettings(install.NumRacksOption(racks))
+		c.Start(ctx, t.L(), startOpts, settings, roachNodes)
+
+		{
+			promCfg := &prometheus.Config{}
+			promCfg.WithPrometheusNode(workloadNode.InstallNodes()[0])
+			promCfg.WithNodeExporter(c.All().InstallNodes())
+			promCfg.WithCluster(roachNodes.InstallNodes())
+			promCfg.WithGrafanaDashboard("http://go.crdb.dev/p/snapshot-admission-control-grafana")
+			promCfg.ScrapeConfigs = append(promCfg.ScrapeConfigs, prometheus.MakeWorkloadScrapeConfig("workload",
+				"/", makeWorkloadScrapeNodes(workloadNode.InstallNodes()[0], []workloadInstance{{nodes: workloadNode}})))
+			_, cleanupFunc := setupPrometheusForRoachtest(ctx, t, c, promCfg, []workloadInstance{{nodes: workloadNode}})
+			defer cleanupFunc()
+		}
+
+		t.Status("installing docker")
+		if err := c.Install(ctx, t.L(), workloadNode, "docker"); err != nil {
+			t.Fatal(err)
+		}
+
+		importC := make(chan struct{})
+		m := c.NewMonitor(ctx, roachNodes)
+		m.Go(func(ctx context.Context) error {
+			const dockerRun = `sudo docker run cockroachdb/tpc-e:latest`
+
+			roachNodeIPs, err := c.InternalIP(ctx, t.L(), roachNodes)
+			if err != nil {
+				return err
+			}
+			roachNodeIPFlags := make([]string, len(roachNodeIPs))
+			for i, ip := range roachNodeIPs {
+				roachNodeIPFlags[i] = fmt.Sprintf("--hosts=%s", ip)
+			}
+
+			t.Status("preparing workload (~4 hours)")
+			c.Run(ctx, workloadNode, fmt.Sprintf("%s --customers=%d --racks=%d --init %s",
+				dockerRun, opts.customers, racks, roachNodeIPFlags[0]))
+			close(importC)
+
+			t.Status("running workload (~4 hours)")
+			activeCustomers := int(float64(opts.customers) * opts.activeRatio)
+			result, err := c.RunWithDetailsSingleNode(ctx, t.L(), workloadNode,
+				fmt.Sprintf("%s --customers=%d --active-customers=%d --racks=%d --duration=%s --threads=%d %s",
+					dockerRun, opts.customers, activeCustomers, racks, runDuration, opts.threads, strings.Join(roachNodeIPFlags, " ")))
+			if err != nil {
+				t.Fatal(err.Error())
+			}
+			t.L().Printf("workload output:\n%s\n", result.Stdout)
+			if strings.Contains(result.Stdout, "Reported tpsE :    --   (not between 80% and 100%)") {
+				return errors.New("invalid tpsE fraction")
+			}
+			return nil
+		})
+		m.Go(func(ctx context.Context) error {
+			// Wait for the import to complete.
+			t.Status("waiting for import to complete")
+			select {
+			case <-importC:
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+
+			t.Status("waiting before creating the index")
+
+			// Wait for the load to run for a while before starting the index build.
+			select {
+			case <-time.After(opts.delayBeforeIndex):
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+
+			t.Status("creating index (~30min)")
+
+			db := c.Conn(ctx, t.L(), 1)
+			_, err := db.ExecContext(ctx, "CREATE INDEX ON tpce.cash_transaction (ct_dts)")
```

Comment:
The same suggestion as the other PR -- use an explicit, unique name, suffixed perhaps by the timestamp this was run at so that repeated runs are made easy.

Reply:
I realized when iterating with that one that not specifying a name gives the flexibility to re-run the identical create index (the system does this automatically) - so its not necessary to pick an explicit name unless you want to drop it in the end. Since I wasn't dropping I didn't pick a name. Let me know if you think I should change this.